src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -253,6 +254,76 @@ public IActionResult Recover([FromBody]WalletRecoveryRequest request)
             }
         }
 
+        /// <summary>
+        /// Recovers a wallet using only the extended public key.
+        /// </summary>
+        /// <param name="request">The object containing the parameters used to recover a wallet.</param>
+        /// <returns></returns>
+        [Route("recover-via-extpubkey")]
+        [HttpPost]
+        public IActionResult RecoverViaExtPubKey([FromBody]WalletExtPubRecoveryRequest request)
+        {
+            Guard.NotNull(request, nameof(request));
+
+            if (!this.ModelState.IsValid)
+            {
+                return BuildErrorResponse(this.ModelState);
+            }
+
+            try
+            {
+                string accountExtPubKey = this.SanitiseExtPubKey(request.ExtPubKey);
+
+                Wallet wallet = this.walletManager.RecoverWallet(request.Name, ExtPubKey.Parse(accountExtPubKey), request.AccountIndex, request.CreationDate);
+
+                // start syncing the wallet from the creation date
+                this.walletSyncManager.SyncFromDate(request.CreationDate);
+
+                return this.Ok();
+            }
+            catch (WalletException e)
+            {
+                // indicates that this wallet already exists
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.Conflict, e.Message, e.ToString());
+            }
+            catch (FileNotFoundException e)
+            {
+                // indicates that this wallet does not exist
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.NotFound, "Wallet not found.", e.ToString());
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
+            }
+        }
+
+        private string SanitiseExtPubKey(string requestExtPubKey)
+        {
+            if (this.network.IsBitcoin())
```

Comment:
Missing entry and exit logs

Reply:
Other methods in there don't have it, so do we really want to do it for these ones ?

Reply:
Yep. Those methods are old and probably created before @Aprogiena introduced logging into the project. So we should start at some point here I guess.